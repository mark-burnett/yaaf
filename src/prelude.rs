pub use crate::{
    actor::{Actor, Tell},
    context::{Context, Publish},
    handler::Handler,
    source::Source,
    system::System,
};
pub use ::async_trait::async_trait;
pub use ::yaaf_macros::{Actor, Source};
