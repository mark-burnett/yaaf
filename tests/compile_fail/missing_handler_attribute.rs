use ::yaaf::prelude::*;

#[derive(Clone, Debug)]
struct ValidMessage;

#[derive(Clone, Debug)]
struct InvalidMessage;

#[derive(Actor)]
#[handle(ValidMessage)]
struct MyActor;

#[async_trait]
impl Handler<ValidMessage> for MyActor {
    async fn handle(&mut self, _ctx: &mut Context<Self>, _message: ValidMessage) {
    }
}

#[async_trait]
impl Handler<InvalidMessage> for MyActor {
    async fn handle(&mut self, _ctx: &mut Context<Self>, _message: InvalidMessage) {
    }
}

#[tokio::main]
async fn main() -> Result<(), Box<dyn ::std::error::Error>> {
    let mut system = System::new();

    let actor = MyActor;

    let address = system.add_actor(actor).await?;

    address.tell(InvalidMessage)?;

    Ok(())
}
