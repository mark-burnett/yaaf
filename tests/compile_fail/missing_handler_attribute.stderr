error[E0277]: the trait bound `MyActor: HandlerRegistered<InvalidMessage>` is not satisfied
  --> $DIR/missing_handler_attribute.rs:20:6
   |
20 | impl Handler<InvalidMessage> for MyActor {
   |      ^^^^^^^^^^^^^^^^^^^^^^^ the trait `HandlerRegistered<InvalidMessage>` is not implemented for `MyActor`
   |
  ::: $WORKSPACE/src/handler.rs
   |
   | pub trait Handler<M: Message>: Actor + HandlerRegistered<M> + Send {
   |                                        -------------------- required by this bound in `yaaf::Handler`
   |
   = help: the following implementations were found:
             <MyActor as HandlerRegistered<ValidMessage>>
